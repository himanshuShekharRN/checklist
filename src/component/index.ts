import {Space} from './space/Space';
import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';

export {
  Card,
  Space,
  Title,
  ListCard,
  CustomSVG,
  CustomIcon,
  SwipableCards,
  RoundedButton,
  LinearProgressBar,
  FloatingAddButton,
  HeadersWithButton,
  SwipableActionButton,
};
