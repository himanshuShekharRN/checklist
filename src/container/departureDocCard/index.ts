import {DepartureDocCard} from './DepartureDocCard';

export {DepartureDocCard};
