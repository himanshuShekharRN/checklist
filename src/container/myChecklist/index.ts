import {MyCheckList} from './MyCheckList';

export {MyCheckList};
