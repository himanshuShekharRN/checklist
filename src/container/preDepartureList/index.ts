import {PreDepartureList} from './PreDepartureList';

export {PreDepartureList};
