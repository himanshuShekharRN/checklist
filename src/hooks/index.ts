import {usePreDepartureListData} from './usePreDepartureListData';
import {useChecklistCompletionStatus} from './useChecklistCompletionStatus';

export {useChecklistCompletionStatus, usePreDepartureListData};
