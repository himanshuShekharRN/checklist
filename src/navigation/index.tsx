import {NavigationContainer} from '@react-navigation/native';
import React from 'react';
import {CheckListStackNavigator} from './checkListStackNavigator';

export const Navigation = () => {
  return (
    <NavigationContainer>
      <CheckListStackNavigator />
    </NavigationContainer>
  );
};
