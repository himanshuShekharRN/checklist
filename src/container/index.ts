import {NationalChecklist} from './nationalChecklist';
import {DepartureDocCard} from './departureDocCard';
import {CustomTab} from './customTab';
import {MyCheckList} from './myChecklist';
import {PreDepartureList} from './preDepartureList';

export {
  NationalChecklist,
  DepartureDocCard,
  CustomTab,
  MyCheckList,
  PreDepartureList,
};
