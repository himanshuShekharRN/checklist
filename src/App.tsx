import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
