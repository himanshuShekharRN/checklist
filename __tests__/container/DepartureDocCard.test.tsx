import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {DepartureDocCard} from '../../src/container';

describe('DepartureDocCard', () => {
  const listData = {
    documentName: 'test',
    nationality: 'RUS',
    docNumber: 87236487,
    issueDate: '20.08.23',
    expiryDate: '31.08.23',
    documentType: 'mandatory',
    status: 'new',
    completed: false,
    isSkipped: false,
  };
  const props = {
    listData: listData,
  };

  let tree: any = render(<DepartureDocCard {...props} />);

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('snapshot created successfully', () => {
    render(<DepartureDocCard {...props} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
