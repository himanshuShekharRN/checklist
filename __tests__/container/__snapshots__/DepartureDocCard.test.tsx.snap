// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepartureDocCard snapshot created successfully 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#CCCDCF",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "paddingHorizontal": 16,
      "paddingVertical": 20,
    }
  }
  testID="departureDocCard"
>
  <View
    testID="customIcon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#005AA5",
            "fontSize": 20,
          },
          undefined,
          {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î¡³
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginLeft": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        test
      </Text>
    </View>
    <Text
      style={
        [
          {
            "marginTop": 2,
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      RUS, 87236487
    </Text>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#707070",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
              "marginTop": 8,
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        Issue date: 
        20.08.23
      </Text>
      <Text
        style={
          [
            {
              "color": "#707070",
              "fontFamily": "Roboto-Regular",
              "fontSize": 12,
              "marginTop": 8,
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        Exp. date: 
        31.08.23
      </Text>
    </View>
  </View>
</View>
`;
